servers/plugins/org.fusesource.ide.server.karaf.core/src/org/fusesource/ide/server/karaf/core/runtime/KarafRuntimeDelegate.java

```diff
@@ -89,6 +89,10 @@ public IStatus validate() {
 			if (!id.toLowerCase().endsWith("karaf.runtime.24")) return new Status(Status.ERROR, Activator.PLUGIN_ID, "Runtime type not compatible with found version...");
         } else if (version != null && version.trim().startsWith("3.0")) {
             if (!id.toLowerCase().endsWith("karaf.runtime.30")) return new Status(Status.ERROR, Activator.PLUGIN_ID, "Runtime type not compatible with found version...");
+        } else if (version != null && version.trim().startsWith("4.0")) {
+            if (!id.toLowerCase().endsWith("karaf.runtime.40")) return new Status(Status.ERROR, Activator.PLUGIN_ID, "Runtime type not compatible with found version...");
```

Comment:
might worth to provide the found version and the runtime type in the message logged

Reply:
might worth extract this duplicated String "Runtime type not compatible with found version..."